### ISSUE TITLE
allow-unrelated-histories

### ISSUE THREAD

**Issue by Dushistov** [jeremysears/scripts#4]
Script `git-submodule-rewrite` always failed for me.
It prints can not merge unrelated histories,
I have to add  `--allow-unrelated-histories` to `git merge --allow-unrelated-histories -s ours --no-commit ${ALLOW_UNRELATED_HISTORIES} "${sub}/master"` to fix this issue.

Hope it would be fixed in main repo also.

**Comment by jeremysears** [comment]
What version of git are you using (git --version)?  This pull request made suppresses the "--allow-unrelated-histories" option, so that the script could support older git versions that don't have a --allow-unrelated-histories option.  It's possible that there's a but in that commit.

https://github.com/jeremysears/scripts/pull/2

**Comment by Dushistov** [comment]
@jeremysears  

git version 2.13.2, I suppose you compare versions in wrong order:
```bash
${OP_VERSION} -le ${GIT_VERSION}
```
mean `if 2.8.4 <= cur_git_version; then ALLOW_UNRELATED_HISTORIES=""`
but you need `if cur_git_version <= 2.8.4; then ALLOW_UNRELATED_HISTORIES=""`

or actually you can remove all version checking and do something like:

```bash
git merge  -s ours --no-commit "${sub}/master" || git merge --allow-unrelated-histories  -s ours --no-commit "${sub}/master"
```

**Comment by jeremysears** [comment]
Fixed in pull request #5
